cpp/compilation-db.rst

```diff
@@ -0,0 +1,28 @@
+
+
+#######################################
+Creating a compilation database
+#######################################
+
+``scons`` does not natively support the creation of a compilation database, instead the
```

Comment:
I think it would be good to add at least a sentence saying what a compilation database is good for.